src/measure_theory/measure_space.lean

```diff
@@ -377,6 +377,25 @@ begin
   exact tendsto_at_top_infi_nat (μ ∘ s) (assume n m hnm, measure_mono $ hm _ _ $ hnm),
 end
 
+/-- The Borel-Cantelli lemma. -/
+lemma measure_limsup_eq_zero {s : ℕ → set α} (hs : ∀ i, is_measurable (s i))
+  (hs' : (∑' i, μ (s i)) ≠ ⊤) : μ (limsup at_top s) = 0 :=
+begin
+  rw limsup_eq_infi_supr_of_nat',
+
+  refine tendsto_nhds_unique
+    (tendsto_measure_Inter (λ i, is_measurable.Union (λ b, hs (b + i))) _
+      ⟨0, lt_of_le_of_lt (measure_Union_le s) (ennreal.lt_top_iff_ne_top.2 hs')⟩)
+    (tendsto_of_tendsto_of_tendsto_of_le_of_le' tendsto_const_nhds
+      (ennreal.tendsto_sum_nat_add (μ ∘ s) hs')
+      (eventually_of_forall (by simp only [forall_const, zero_le]))
+      (eventually_of_forall (λ i, measure_Union_le _))),
```

Comment:
Could you split this gigantic line, to help the reader who would want to follow the proof by following the evolution of the goal? For instance, you could start with
```lean
refine tendsto_nhds_unique
    (tendsto_measure_Inter (λ i, is_measurable.Union (λ b, hs (b + i))) _
      ⟨0, lt_of_le_of_lt (measure_Union_le s) (ennreal.lt_top_iff_ne_top.2 hs')⟩) _,
```
and then prove separately the two remaining goals.

As an aside, in general we don't leave empty lines in the proof. If you feel the need for such a line, you can often feel it with a comment, which will be even more helpful :-)